test/parse/TestScriptParser.cpp

```diff
@@ -0,0 +1,199 @@
+#include <array>
+#include <boost/test/tools/old/interface.hpp>
+#include <boost/test/unit_test.hpp>
+#include <memory>
+
+#include "parse/Parse.h"
+#include "universe/Condition.h"
+#include "universe/Conditions.h"
+#include "universe/Effect.h"
+#include "universe/Effects.h"
+#include "universe/Tech.h"
+#include "universe/UnlockableItem.h"
+#include "universe/ValueRefs.h"
+#include "util/Directories.h"
+#include "util/GameRules.h"
+#include "util/Pending.h"
+
+#include "ParserAppFixture.h"
+
+BOOST_FIXTURE_TEST_SUITE(TestScriptParser, ParserAppFixture)
+
+BOOST_AUTO_TEST_CASE(parse_game_rules) {
+    Pending::Pending<GameRules> game_rules_p = Pending::StartParsing(parse::game_rules, m_scrpiting_dir / "game_rules.focs.txt");
+    auto game_rules = *Pending::WaitForPendingUnlocked(std::move(game_rules_p));
+    BOOST_REQUIRE(!game_rules.Empty());
+    BOOST_REQUIRE(game_rules.RuleExists("RULE_HABITABLE_SIZE_MEDIUM"));
+    BOOST_REQUIRE(GameRules::Type::TOGGLE == game_rules.GetType("RULE_ENABLE_ALLIED_REPAIR"));
+}
+
+BOOST_AUTO_TEST_CASE(parse_techs) {
+    Pending::Pending<TechManager::TechParseTuple> techs_p = Pending::StartParsing(parse::techs<TechManager::TechParseTuple>, m_scrpiting_dir / "techs");
+    auto techs_tuple = *Pending::WaitForPendingUnlocked(std::move(techs_p));
```

Comment:
`auto&& [techs, tech_categories, categories_seen] = *Pending::WaitForPendingUnlocked(std::move(techs_p));`

Reply:
something weird is happening here. for me, the returned tuple has an empty container for index 0, then two invalid references for the other two indicies.

Reply:
Yes, without this change tests are passed but after it applied I get 

```
error: fatal error: in "TestScriptParser/parse_techs": critical check 1 == tech_categories.count("PRODUCTION_CATEGORY") has failed [1 != 0]
```

Reply:
It works with

```cpp
auto [techs, tech_categories, categories_seen] = *Pending::WaitForPendingUnlocked(std::move(techs_p));
```